### ISSUE TITLE
tokenGetter of JWT_OPTIONS is called always

### ISSUE THREAD

**Issue by junalmeida** [auth0/angular2-jwt#551]
The tokenGetter delegate of JWT_OPTIONS is being called even if the request is on blockedRoutes (or not on white list domains).
This is bad because I am using the tokenGetter to check for expired token and refresh it, anda second httpcall pass again at the tokenGetter leading to infinite requests. 

There is no need to call a tokenGetter if the request is not using a token at all.

**Comment by IlCallo** [comment]
Same problem here.
I pinpointed the problem in `jwtService.isTokenExpired(token)` having as default value `token = this.tokenGetter()`.
When the token is `undefined` (maybe also with null), the getter fires, creating the infinite chain of requests, given the fact that the getter contains another call etc etc.
In my case I simply resolved by putting `if (accessToken && jwtService.isTokenExpired(accessToken))`.

This has probably been done to use `.isTokenExpired()` without arguments as shortcut without having to retrieve the actual token as specified [here](https://github.com/auth0/angular2-jwt#istokenexpired-old-tokennotexpired-function).

Given the fact that checking for token expiration inside tokenGetter seems a common practice, maybe it would be a good idea to remove this shortcut?
Or at least specify this problem in the docs
